src/FormHandler/UserFormHandler.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace BiffBangPow\MessageBoard\FormHandler;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use BiffBangPow\MessageBoard\Model\User;
+use BiffBangPow\MessageBoard\Services\PasswordEncryptionService;
+use BiffBangPow\MessageBoard\Services\SessionService;
+
+class UserFormHandler
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+    /**
+     * @var SessionService
+     */
+    private $sessionService;
+
+    /**
+     * UserFormHandler constructor.
+     * @param EntityManagerInterface $entityManager
+     * @param SessionService $sessionService
+     */
+    public function __construct(EntityManagerInterface $entityManager, SessionService $sessionService)
+    {
+        $this->entityManager = $entityManager;
+        $this->sessionService = $sessionService;
+    }
+
+    public function handleRegistration(Request $request)
+    {
+        $passwordEncryptionServices = new PasswordEncryptionService();
+        $user = new User();
+
+        $username = ($request->get('username'));
+        $passwordInput = ($request->get('password'));
+
+        $user->setUsername($username);
+        $user = $passwordEncryptionServices->encryptPassword($user, $passwordInput);
+
+        $this->entityManager->persist($user);
+        $this->entityManager->flush();
+
+        $this->sessionService->setUserId($user->getId());
+    }
+
+    /**
+     * @param Request $request
+     * @return bool
+     */
+    public function handleLogin(Request $request)
+    {
+        $username = ($request->get('username'));
+        $passwordInput = ($request->get('password'));
+
+        $passwordDatabase =  $this->entityManager->createQuery('Select u.password from BiffBangPow\MessageBoard\Model\USER  u WHERE u.username = ?1')
```

Comment:
this works fine but using the query builder makes much nicer code. It would be _something_ like this:
```
$user = $this->userRepository
    ->getQueryBuilder('u')
    ->where('u.username = :username')
    ->setParameter('username', $userName)
    ->getQuery()
    ->execute()
;
````
I haven't tested that so might not be 100% but here is the documentation: http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/query-builder.html